src/pushsource/_impl/backend/pub_source/pub_source.py

```diff
@@ -0,0 +1,108 @@
+import logging
+from concurrent import futures
+
+from more_executors import Executors
+
+from ...helpers import (
+    as_completed_with_timeout_reset,
+    force_https,
+    list_argument,
+    try_int,
+)
+from ...model import AmiPushItem
+from ...source import Source
+from .pub_client import PubClient
+
+LOG = logging.getLogger("pushsource")
+
+
+class PubSource(Source):
+    """Uses push item list from Pub as the source of push items."""
+
+    def __init__(
+        self,
+        url,
+        task_id,
+        threads=4,
+        timeout=60 * 60 * 4,
+    ):
+        """Create a new source.
+
+        Parameters:
+            url (src)
+                Base URL of Pub Tool, e.g. "http://pub.example.com",
+                "https://pub.example.com:8123".
+
+            task_id (int, str, list[str], list[int])
+                Task ID(s) to be used as push item source.
+                If a single string is given, multiple IDs may be
+                comma-separated.
+
+            threads (int)
+                Number of threads used for concurrent queries to Errata Tool
+                and koji.
```

Comment:
ET and koji are not involved, so it should say either "to Pub" or maybe just not specify any system.

```suggestion
                Number of threads used for concurrent queries.
```